src/core_read_parameter_file.c

```diff
@@ -2,217 +2,260 @@
     ParamAddr[NParam] = &(run_params->ThreshMajorMerger);
     ParamID[NParam++] = DOUBLE;
 
-    strcpy(ParamTag[NParam], "RecycleFraction");
+    strncpy(ParamTag[NParam], "RecycleFraction", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->RecycleFraction);
     ParamID[NParam++] = DOUBLE;
 
-    strcpy(ParamTag[NParam], "ReIncorporationFactor");
+    strncpy(ParamTag[NParam], "ReIncorporationFactor", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->ReIncorporationFactor);
     ParamID[NParam++] = DOUBLE;
 
-    strcpy(ParamTag[NParam], "UnitVelocity_in_cm_per_s");
+    strncpy(ParamTag[NParam], "UnitVelocity_in_cm_per_s", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->UnitVelocity_in_cm_per_s);
     ParamID[NParam++] = DOUBLE;
 
-    strcpy(ParamTag[NParam], "UnitLength_in_cm");
+    strncpy(ParamTag[NParam], "UnitLength_in_cm", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->UnitLength_in_cm);
     ParamID[NParam++] = DOUBLE;
 
-    strcpy(ParamTag[NParam], "UnitMass_in_g");
+    strncpy(ParamTag[NParam], "UnitMass_in_g", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->UnitMass_in_g);
     ParamID[NParam++] = DOUBLE;
 
-    strcpy(ParamTag[NParam], "Hubble_h");
+    strncpy(ParamTag[NParam], "Hubble_h", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->Hubble_h);
     ParamID[NParam++] = DOUBLE;
 
-    strcpy(ParamTag[NParam], "ReionizationOn");
+    strncpy(ParamTag[NParam], "ReionizationOn", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->ReionizationOn);
     ParamID[NParam++] = INT;
 
-    strcpy(ParamTag[NParam], "SupernovaRecipeOn");
+    strncpy(ParamTag[NParam], "SupernovaRecipeOn", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->SupernovaRecipeOn);
     ParamID[NParam++] = INT;
 
-    strcpy(ParamTag[NParam], "DiskInstabilityOn");
+    strncpy(ParamTag[NParam], "DiskInstabilityOn", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->DiskInstabilityOn);
     ParamID[NParam++] = INT;
 
-    strcpy(ParamTag[NParam], "SFprescription");
+    strncpy(ParamTag[NParam], "SFprescription", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->SFprescription);
     ParamID[NParam++] = INT;
 
-    strcpy(ParamTag[NParam], "AGNrecipeOn");
+    strncpy(ParamTag[NParam], "AGNrecipeOn", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->AGNrecipeOn);
     ParamID[NParam++] = INT;
 
-    strcpy(ParamTag[NParam], "BaryonFrac");
+    strncpy(ParamTag[NParam], "BaryonFrac", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->BaryonFrac);
     ParamID[NParam++] = DOUBLE;
 
-    strcpy(ParamTag[NParam], "Omega");
+    strncpy(ParamTag[NParam], "Omega", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->Omega);
     ParamID[NParam++] = DOUBLE;
 
-    strcpy(ParamTag[NParam], "OmegaLambda");
+    strncpy(ParamTag[NParam], "OmegaLambda", MAXTAGLEN);
     ParamAddr[NParam] = &(run_params->OmegaLambda);
     ParamID[NParam++] = DOUBLE;
 
-    strcpy(ParamTag[NParam], "PartMass");
+    strncpy(ParamTag[NParam], "PartMass", MAXTAGLEN);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Easily used incorrectly; doesn't always \0-terminate or check for invalid pointers [MS-banned] (CWE-120).  Risk is low because the source is a constant string.](https://app.codacy.com/manual/jacobseiler/sage-model/pullRequest?prid=4620074)